packages/split-layout/src/vaadin-split-layout-mixin.js

```diff
@@ -63,17 +63,34 @@ export const SplitLayoutMixin = (superClass) =>
 
     /** @private */
     _processChildren() {
-      [...this.children].forEach((child, i) => {
-        if (i === 0) {
-          this._primaryChild = child;
-          child.setAttribute('slot', 'primary');
-        } else if (i === 1) {
-          this._secondaryChild = child;
-          child.setAttribute('slot', 'secondary');
-        } else {
-          child.removeAttribute('slot');
-        }
-      });
+      const children = [...this.children];
```

Comment:
**nit:** The method could be split into two.

```js
[...this.children].forEach((child) => {
  if (child.hasAttribute('slot')) {
    this._processChildWithSlot(child);
  } else {
    this._processChildWithoutSlot(child);
  }
});
```

Reply:
Added the methods. I still need to have two loops, because in the first one, I need to do some cleanup before trying to assign the correct slots to any remaining child without a slot.